airflow/utils/dag_processing.py

```diff
@@ -691,6 +688,7 @@ def start(self):
                               "have been processed %s times", self._max_runs)
                 break
 
+            # TODO can this be removed?
```

Comment:
@tooptoop4 Please don't add comments like this  -- it just makes it harder and confusing for the next person who comes along.

The answer is no - async mode is still needed -- when using SQLlite we run "synchronously" as it doesn't have well/at all with multiple access, so we want to "stop" as soon as we're done, not sleep. poll_time=None in sync mode, which says wait until a message is sent on L639.

Please remove this comment.

Reply:
@ashb async mode is not sqllite. so i am questioning why non-sqllite need that IF block at all. ie can non-sqllite use original poll time set on L626

Reply:
Async mode is set to true when sqlite is in use.

Reply:
@ashb I think u are confused here, see https://github.com/apache/airflow/blob/master/airflow/jobs/scheduler_job.py#L1485
# When using sqlite, we do not use async_mode
        # so the scheduler job and DAG parser don't access the DB at the same time.
        async_mode = not self.using_sqlite

Reply:
Whops yes totally backwards. Sqlite is sync.

Reply:
But no, I don't think this can be removed -- the default timeout is 0 which means this would sit in a busy/CPU consuming loop.